cmd/dep/testdata/harness_tests/init/case4/final/Gopkg.toml

```diff
@@ -0,0 +1,8 @@
+
+[[constraint]]
+  name = "github.com/sdboyer/deptest"
+  version = "1.0.0"
```

Comment:
Yup, when init generates the manifest it should not write out the `^`. When the manifest is read, the caret is automatically added. Here's some [backstory on why we don't include the caret in the manifest](https://github.com/golang/dep/pull/631#issuecomment-303589618).

My preference is for the verbose output to print what's _really_ going on and show the caret, but will leave that call to @sdboyer. If we do end up changing that, let's do it in a separate PR.